packages/integration-tests/src/retry.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * Executes an asynchronous operation with retry logic.
+ * This function attempts to execute the provided callable function multiple times
+ * based on the specified retry conditions. It's useful for handling transient
+ * errors or temporary service unavailability.
+ */
+export const runWithRetry = async <T>(
+  callable: () => Promise<T>,
+  retryPredicate: (error: Error) => boolean,
+  maxAttempts = 3
+): Promise<T> => {
+  const collectedErrors: Error[] = [];
+
+  for (let attempt = 1; attempt <= maxAttempts; attempt++) {
+    try {
+      const result = await callable();
+      return result;
+    } catch (error) {
+      if (error instanceof Error) {
+        collectedErrors.push(error);
+
+        if (!retryPredicate(error) || attempt === maxAttempts) {
+          break;
```

Comment:
```suggestion
          throw error;
```

It turns out that we shouldn't be throwing aggregate error if predicate doesn't match. The output is confusing.

![image](https://github.com/user-attachments/assets/7ee4c209-0a78-42d3-b38b-6095221d3d4e)